nextcord/message.py

```diff
@@ -543,6 +545,56 @@ def flatten_handlers(cls):
     return cls
 
 
+class MessageInteraction:
+    """Represents a message's interaction data.
+
+    A message's interaction data is a property of a message when the message
+    is a response to an interaction from any bot.
+
+    .. versionadded:: 2.0
+
+    Attributes
+    -----------
+    data: :class:`dict`
+        The raw data from the interaction.
+    id: :class:`int`
+        The interaction's ID.
+    type: :class:`InteractionType`
+        The interaction type.
+    name: :class:`str`
+        The name of the application command.
+    user: Union[:class:`User`, :class:`Member`]
+        The :class:`User` who invoked the interaction or :class:`Member` if the interaction
+        occurred in a guild.
+    """
+
+    __slots__ = (
+        "_state",
+        "data",
+        "id",
+        "type",
+        "name",
+        "user",
+    )
+
+    def __init__(
+        self, *, data: MessageInteractionPayload, guild: Optional[Guild], state: ConnectionState
+    ):
+        self._state: ConnectionState = state
+
+        self.data: MessageInteractionPayload = data
+        self.id: int = int(data["id"])
+        self.type: int = data["type"]
+        self.name: str = data["name"]
+        if "member" in data and guild is not None:
+            self.user = Member(state=self._state, guild=guild, data={**data["member"], "user": data["user"]})  # type: ignore
+        else:
+            self.user = self._state.create_user(data=data["user"])
+
+    def __repr__(self):
+        return f"<{self.__class__.__name__} id={self.id} type={self.type} name={self.name} user={self.user!r}>"
+
```

Comment:
As this object has a unique snowflake ID, we should be able to turn this into a created_at timetamp like most other objects. This time will be different from the message since the interaction occurs before the message is sent.

```suggestion

    @property
    def created_at(self) -> datetime.datetime:
        """:class:`datetime.datetime`: The interaction's creation time in UTC."""
        return snowflake_time(self.id)
```

Reply:
Interesting, never knew that